File: src/main/java/net/hectus/neobb/turn/person_game/structure/PTPumpkinWall.java
package net.hectus.neobb.turn.person_game.structure;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import net.hectus.neobb.structure.Structure;
import net.hectus.neobb.structure.StructureManager;
import net.hectus.neobb.turn.default_game.structure.StructureTurn;
import net.hectus.neobb.turn.person_game.categorization.UtilityCategory;
import net.hectus.neobb.util.MinecraftTime;

public class PTPumpkinWall extends StructureTurn implements UtilityCategory {
    public PTPumpkinWall(NeoPlayer player) { super(player); }
    public PTPumpkinWall(PlacedStructure data, NeoPlayer player) { super(data, player); }

    @Override
    public Structure referenceStructure() {
        return StructureManager.structure("pumpkin_wall");
    }

    @Override
    public void apply() {
        super.apply();
        player.game.time(MinecraftTime.MIDNIGHT);
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/structure/PTPumpkinWall.kt
package net.hectus.neobb.turn.person_game.structure

import com.marcpg.libpg.storing.Cord
import com.marcpg.libpg.util.MinecraftTime
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures
import net.hectus.neobb.turn.default_game.structure.StructureTurn
import net.hectus.neobb.turn.person_game.categorization.UtilityCategory

class PTPumpkinWall(data: PlacedStructure?, cord: Cord?, player: NeoPlayer?) : StructureTurn(data, cord, player), UtilityCategory {
    override val staticStructure: StaticStructure = StaticStructures.Person.PUMPKIN_WALL

    override fun apply() {
        player!!.game.time = MinecraftTime.MIDNIGHT
    }
}
